Prefer Github/Gitlab environment variables to get owner/project

ziouf opened the pull request:
related to #363

relekang reviewed (changes requested):
Please fix the test by mocking the environment variable and add a test for using the environment variable.

ziouf commented:
> 
> 
> Please fix the test by mocking the environment variable and add a test for using the environment variable.

Added some tests and updated upstream

ziouf commented:
Could you review again this pull request ?
Thank you

relekang commented on semantic_release/vcs_helpers.py:
```diff
@@ -19,7 +20,8 @@
 except InvalidGitRepositoryError:
-    repo = None
+    # repo = None
+    raise GitError("Not in a valid git repository")
```
Why do we need to raise an error here?

ziouf replied:
does it make sense to run python-semantic-release outside of a valid git repository ?
I agree it could be managed diffrently. 
Generally I prefer an error message followed by exit rather noop.

relekang replied:
This code runs on import as a part of a command and thus buy passes the error handling.

ziouf replied:
@relekang mypy is failing because of your initial implement
 
https://github.com/relekang/python-semantic-release/runs/4270162688?check_suite_focus=true#step:5:8

Maybe you need to change error handling implement. Or not initialize repo variable in global scope ...

fleXible replied:
instead of assigning None a simple pass will do, because it will set repo to None implicitly and mypy won't complain. And it's safe to do that, because the `check_repo' wrapper checks for repo for None and raises an exception.
Instead of using a global variable, it would be better to just wrap the whole thing in a class.

relekang commented:
Thanks, merged it locally to fix the conflicts
